Source/WebCore/dom/Element.cpp

```diff
@@ -2603,6 +2614,12 @@ Node::InsertedIntoAncestorResult Element::insertedIntoAncestor(InsertionType ins
             ensureElementRareData().setEffectiveLang(lang);
     }
 
+    if (!UNLIKELY(hasRareData()) || !elementRareData()->effectiveLang()) {
+        auto langAttr = langFromAttribute();
+        if (!langAttr.isNull() && langAttr != document().documentElementLanguage())
```

Comment:
This doesn't look right. Why does ancestor of this element changing anything to do with effective lang getting changed? If we're trying to detect when an element is moved from one document to another, then we should be doing that in Element::didMoveToNewDocument instead.